### DISCUSSION TITLE
basePath and i18n support

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#22433] [Help]
Wondering if there is a targeted date to support the i18n features with applications that have to use basePath:

https://github.com/vercel/next.js/discussions/17078?sort=old#discussioncomment-107903

Currently this is a struggle for us as we have 3 solutions, but all the solutions take use away from being able to use 'next export'

1. Remove all SSG and just use SSR to access the request to get the locale and submit that to our Headless CMS.
2. Use rewrites to get access to url before the context path (basePath).  But we also see an issue in Next if you do rewrites with basePath and url params where the basePath shows up twice in a url.  Plus the fact that users will now see the rewrite in URLs (so we have to resolve this).
3. We remove basePath all together.  Then have to manually add it back to links, assetPrefix, buildId, etc. and then create a bunch of rewrite rules for _next files as well as modify our reverse proxy for all of these routes.

We are pushing towards 3, but as mentioned, all solutions take us away from 'next export'.

Any target for Next to support basePath/i18n?
Or anyone else solved this another way that would support 'next export'?

Thanks.

**Comment by robertwbradford** [comment]
https://github.com/vercel/next.js/discussions/17078#discussioncomment-220722 is one idea for a public API on this.
